dot/sync/chain_sync.go

```diff
@@ -582,12 +615,16 @@ func (cs *chainSync) handleWorkersResults(
 	workersResults chan *syncTaskResult, origin blockOrigin, startAtBlock uint, expectedSyncedBlocks uint32) error {
 	startTime := time.Now()
 
+	// TODO one idea is can refactor this to be a map (will be not as efficient tho, so prob not desired)
```

Comment:
I don't think using a map here can bring benefits, one of the reasons is that it should ensure the order of the synced blocks and maps does not hold this constraint.

Reply:
yeah i actually meant to delete this for exactly this reason, will update